sdk/identity/azure-identity/test/ut/client_certificate_credential_test.cpp

```diff
@@ -148,6 +148,77 @@ TEST_P(GetCredentialName, )
   EXPECT_EQ(cred.GetCredentialName(), "ClientCertificateCredential");
 }
 
+TEST(ClientCertificateCredential, UnsupportedExtension)
+{
+  try
+  {
+    ClientCertificateCredential const cred(
+        "01234567-89ab-cdef-fedc-ba8976543210", "fedcba98-7654-3210-0123-456789abcdef", "file.pfx");
+
+    EXPECT_TRUE(
+        !"ClientCertificateCredential with unsupported extension (.pfx) is supposed to throw.");
```

Comment:
What does this `!` operator supposed to mean here on a string, when passed to `EXPECT_TRUE`?

Whatever the intent is here, I think we should write it clearly, so it's easier to understand.

Reply:
This is a way to fail with message, it is kind of idiomatic - there is C and C++ code that uses it with `assert()`s, for example.
The way it works is that any string literal is never null, but `!` inverts that, i.e. `!""` is always false, and so when we `EXPECT_TRUE()` or `assert()`, the assertion always fails.